strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetPackageGenerator.java

```diff
@@ -46,20 +50,47 @@
             "<Relationship Type=\"http://schemas.openxmlformats.org/package/2006/relationships/metadata/core-properties\" Target=\"/package/services/metadata/core-properties/metadata.psmdcp\" Id=\"R23f62\r\ne2778b3442e\" />\r\n"  +
             "</Relationships>";
     
-    private String basedir;
+    private Path basedir;
 
     public NugetPackageGenerator()
     {
     }
 
     public NugetPackageGenerator(String basedir)
     {
-        this.basedir = basedir;
+        this.basedir = Paths.get(basedir);
     }
 
     public NugetPackageGenerator(File basedir)
     {
-        this.basedir = basedir.getAbsolutePath();
+        this.basedir = basedir.toPath();
+    }
+
+    public NugetPackageGenerator(Path basedir)
+    {
+        this.basedir = basedir;
+    }
+
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
+            throws IOException
+    {
+        NugetArtifactCoordinates coordinates = NugetArtifactCoordinates.parse(Paths.get(uri).toString());
```

Comment:
@yugander-krishan-singh can you please use just `uri.toString()` as was requested before.